pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -203,6 +203,9 @@ def new_process(self, process_idx, trainer, mp_queue):
         # persist info in ddp_spawn
         self.transfer_distrib_spawn_state_on_fit_end(results)
 
+        # https://github.com/PyTorchLightning/pytorch-lightning/pull/8217#issuecomment-871261949
+        trainer._call_teardown_hook(self.lightning_module)
+
```

Comment:
Please add some text comments in addition to the link to the GitHub issue. it'll be easier to read rather than needing ro follow links

something like
```
# Ensure that spawned processes go through teardown before joining
# for more context, see <GitHub link> 
```